Sources/AppCoinsSDK/Domain/Repositories/Currency/CurrencyRepository.swift

```diff
@@ -54,7 +54,7 @@ internal class CurrencyRepository: CurrencyRepositoryProtocol {
             case .success(let supportedCurrencies):
                 if let supportedCurrency = supportedCurrencies.first(where: { $0.currency == currency }) {
                     completion(.success(supportedCurrency))
-                } else { completion(.failure(.failed)) }
+                } else { completion(.failure(.failed(message: "getSupportedCurrency method Failed", description: "Unsupported currency: \(currency)", request: nil))) }
```

Comment:
Message should not be technical, the description should. 

One example of how this could be resolved is:
- message: "Failed to get supported currency"
- description: "Received unsupported currency: \(currency) at CurrencyRepository.swift:getSupportedCurrency"

Reply:
Done